src/Func.h

```diff
@@ -100,8 +110,15 @@ class Func : public Obj
 		}
 
 	// Add a new event handler to an existing function (event).
-	virtual void AddBody(detail::StmtPtr new_body, const std::vector<detail::IDPtr>& new_inits,
-	                     size_t new_frame_size, int priority = 0);
+	[[deprecated("Remove in v6.1 - Use function_ingredients version")]] virtual void
+	AddBody(detail::StmtPtr new_body, const std::vector<detail::IDPtr>& new_inits,
+	        size_t new_frame_size, int priority = 0);
+
+	// Add a new event handler to an existing function (event).
+	virtual void AddBody(const zeek::detail::function_ingredients* ingredients);
```

Comment:
Use a smart pointer for the argument here to make ownership/memory management clear.

Reply:
Thanks. I'm heavily considering removing this part and adding an overload/optional parameter for the additional groups instead. I'm not entirely happy with either one, but leaning towards less happy with the new `function_ingredients` parameter due to changing the additional signature and deprecation (at the cost of not being happy adding yet another parameter...).

Reply:
Agree, I actually almost suggested that as well.